packages/gensx-core/src/checkpoint.ts

```diff
@@ -26,11 +26,11 @@ export function generateDeterministicId(
   // Simple but effective serialization for MVP
   const propsStr = JSON.stringify(props, Object.keys(props).sort());
   const propsHash = createHash("sha256")
-    .update(propsStr)
+    .update(`${propsStr}:${parentId ?? "root"}`)
     .digest("hex")
     .slice(0, 16);
 
-  return `${parentId ?? "root"}:${name}:${propsHash}`;
+  return `${name}:${propsHash}`;
```

Comment:
If we include `parentId` in the ID, IDs will get arbitrarily long, relative to the nesting.

Reply:
This is true.

The impact of including 
1. "Identity" is stronger - fewer surprises when components that match props but not parent get skipped during checkpoint replay
2. checkpoints get (exponentially) bigger the more you nest
3. Not practical to store these IDs in a database unless we hash them -- not sure if we need to. 

Are you comfortable making this simplification? Without framework limits, we'll probably still have problems storing the name-props-only version of this ID in the database without hashing.